File: fix_json.js
const fs = require('fs');
let content = fs.readFileSync('./mq-test/analysis.json', 'utf8');

// Fix string concatenation
content = content.replace(/" \+ "/g, "");

// Fix the missing bracket in line 145: "friendship": { ... "desc": "...", "life": { ... }
// We need to change `", "life": {` to `"}, "life": {`
content = content.replace(/유쾌한 리더입니다.", "life"/g, '유쾌한 리더입니다."}, "life"');

fs.writeFileSync('./mq-test/analysis.json', content);
console.log('Fixed analysis.json');

File: fix_json.py
import re

with open('sq-test/analysis.json', 'r', encoding='utf-8') as f:
    data = f.read()

# Replace string concatenation
data = re.sub(r'"\s*\+\s*"', '', data)

# The pattern is: ", "religion": or ", "allergy": etc on the SAME LINE as a previous key's desc.
data = re.sub(r'("\s*,\s*")((?:religion|allergy|wellness|spending|prescription)":\s*\{)', r'"} ,\n    "\2', data)

with open('sq-test/analysis.json', 'w', encoding='utf-8') as f:
    f.write(data)
